adam/experiment/observer.py

```diff
@@ -803,9 +805,31 @@ def observe(  # pylint: disable=unused-argument
         experiment_dir.mkdir(parents=True, exist_ok=True)
 
         if self.copy_curriculum and isinstance(situation, SimulationSituation):
+
+            if isinstance(perceptual_representation,VisualPerceptionRepresentation):
+                frames = perceptual_representation.frames
+                imagesFiles = situation.scene_images_png
+                for index, imageFrame in enumerate(zip(frames,imagesFiles)):
```

Comment:
nit: As the `frames` and `imageFiles` variables aren't reused just pass the original references into zip.
```suggestion
                for index, imageFrame in enumerate(zip(perceptual_representation.frames, situation.scene_images_png)):
```